pkg/networkserver/grpc_asns.go

```diff
@@ -310,7 +319,10 @@ func (ns *NetworkServer) DownlinkQueueReplace(ctx context.Context, req *ttnpb.Do
 
 // DownlinkQueuePush is called by the Application Server to push a downlink to queue for a device.
 func (ns *NetworkServer) DownlinkQueuePush(ctx context.Context, req *ttnpb.DownlinkQueueRequest) (*pbtypes.Empty, error) {
-	if len(req.Downlinks) == 0 {
+	switch n := len(req.Downlinks); {
+	case uint64(n) > ns.downlinkQueueCapacity:
+		return nil, errDownlinkQueueCapacityExceeded.New()
+	case n == 0:
 		return ttnpb.Empty, nil
 	}
```

Comment:
More readable and shorter version:

```suggestion
	if n := len(req.Downlinks); uint64(n) > ns.downlinkQueueCapacity {
		return nil, errDownlinkQueueCapacityExceeded.New()
	} else if n == 0 {
		return ttnpb.Empty, nil
	}
```